src/AbstractTwigCallable.php

```diff
@@ -33,10 +33,35 @@ public function __construct(string $name, $callable = null, array $options = [])
             'needs_context' => false,
             'needs_charset' => false,
             'is_variadic' => false,
+            'deprecation_info' => null,
             'deprecated' => false,
             'deprecating_package' => '',
             'alternative' => null,
         ], $options);
+
+        if ($this->options['deprecation_info'] && !$this->options['deprecation_info'] instanceof DeprecatedCallableInfo) {
+            throw new \LogicException(\sprintf('The "deprecation_info" option must be an instance of "%s".', DeprecatedCallableInfo::class));
+        }
+
+        if ($this->options['deprecated']) {
+            if ($this->options['deprecation_info']) {
+                throw new \LogicException('When setting the "deprecation_info" option, you need to remove the obsolete deprecated options.');
```

Comment:
The option `deprecated` could accept an instance of `DeprecatedCallableInfo` so that you don't have to introduce a new option. The object being truthy, that is compatible with older versions.

Reply:
Not sure, I like the new name, it's more descriptive.